cpp/include/ucxx/endpoint.h

```diff
@@ -83,10 +53,15 @@ class Endpoint : public Component {
   ucp_ep_h _originalHandle{nullptr};  ///< Handle to the UCP endpoint, after it was previously
                                       ///< closed, used for logging purposes only
   bool _endpointErrorHandling{true};  ///< Whether the endpoint enables error handling
-  std::unique_ptr<ErrorCallbackData> _callbackData{
-    nullptr};  ///< Data struct to pass to endpoint error handling callback
-  std::shared_ptr<InflightRequests> _inflightRequests{
-    std::make_shared<InflightRequests>()};  ///< The inflight requests
+  std::unique_ptr<InflightRequests> _inflightRequests{
+    std::make_unique<InflightRequests>()};  ///< The inflight requests
+  std::mutex _mutex{std::mutex()};          ///< Mutex used to prevent race conditions with
+                                            ///< `ucxx::Endpoint::setCloseCallback()`.
```

Comment:
*nit*: Document what this mutex protects?

Reply:
Does https://github.com/rapidsai/ucxx/pull/238/commits/8c3148dd72f5ca7d599b9949c385a77d783edaa8 suits your expectations here?